README
LICENSE

  Permission to use, copy, modify, and distribute this software and its
documentation for any purpose and without fee is hereby granted, provided
that the above copyright notice appears in all copies and that both the
copyright notice and this permission notice appear in supporting
documentation, and that the same name not be used in advertising or
publicity pertaining to distribution of the software without specific,
written prior permission. We make no representations about the
suitability this software for any purpose. It is provided "as is"
without express or implied warranty.

CREDITS

  xzdec: XZ Utils,
Released under the public domain.
(http://tukaani.org/xz/)

  sort: coreutils, Copyright (C) 1988, 1991-2011 Free Software Foundation, Inc.,
licensed under the GNU General Public License (http://www.gnu.org/licenses/gpl.html).
(http://www.gnu.org/software/coreutils)

  shlock: shlock.c, Copyright (c) 2005 The DragonFly Project,
licensed under the DragonFly BSD License (http://www.dragonflybsd.org/docs/developer/DragonFly_BSD_License/).
(http://gitweb.dragonflybsd.org/dragonfly.git/blob/HEAD:/usr.bin/shlock/shlock.c)

  convert, identify, mogrify: ImageMagick, Copyright (C) 1999-2011 ImageMagick Studio LLC,
licensed under the Apache 2.0 license (http://www.imagemagick.org/script/license.php)
(http://www.imagemagick.org/script/index.php)

    |->   libpng, Copyright (c) 2004, 2006-2012 Glenn Randers-Pehrson,
    |   Licensed under the libpng license (http://www.libpng.org/pub/png/src/libpng-LICENSE.txt)
    |   (http://www.libpng.org/pub/png/libpng.html)
    |
    |->   zlib, Copyright (C) 1995-2013 Jean-loup Gailly and Mark Adler,
    |   Licensed under the zlib license (http://zlib.net/zlib_license.html)
    |   (http://zlib.net/)
    |
    `->   libjpeg-turbo, Copyright (C) 1991-2010, Thomas G. Lane, Guido Vollbeding.,
        Licensed under a BSD-style license (http://sourceforge.net/p/libjpeg-turbo/code/HEAD/tree/trunk/README).
        (http://libjpeg-turbo.virtualgl.org/)

  inotifywait: inotify-tools, Copyright (C) 2006-2012 Rohan McGovern,
licensed under the GNU General Public License, version 2 (http://www.gnu.org/licenses/old-licenses/gpl-2.0.html).
(https://github.com/rvoicilas/inotify-tools)

  libfreetype.so: FreeType, Copyright 1996-2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011 by David Turner, Robert Wilhelm, and Werner Lemberg,
Licensed under the FreeType License (http://freetype.org/FTL.TXT). (http://freetype.org)

  libharfbuzz.so: HarfBuzz, Copyright © 2010,2011,2012  Google, Inc., Copyright © 2012  Mozilla Foundation, Copyright © 2011  Codethink Limited, Copyright © 2008,2010  Nokia Corporation and/or its subsidiary(-ies),
    Copyright © 2009  Keith Stribley, Copyright © 2009  Martin Hosken and SIL International, Copyright © 2007  Chris Wilson, Copyright © 2006  Behdad Esfahbod, Copyright © 2005  David Turner,
    Copyright © 2004,2007,2008,2009,2010  Red Hat, Inc., Copyright © 1998-2004  David Turner and Werner Lemberg,
Licensed in part under the "Old MIT" license (http://cgit.freedesktop.org/harfbuzz/tree/COPYING)
(http://www.freedesktop.org/wiki/Software/HarfBuzz/)


ChangeLog.txt
2020-06-13 20:55  NiLuJe

	* [r17508] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libturbojpeg.so.0,
	  src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-06-05 00:39  NiLuJe

	* [r17466] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libharfbuzz.so.0, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-06-02 01:48  NiLuJe

	* [r17441] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-05-18 20:05  NiLuJe

	* [r17349] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libharfbuzz.so.0, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-05-13 02:19  NiLuJe

	* [r17279] src/linkss/bin/convert, src/linkss/bin/identify,
	  src/linkss/bin/mogrify, src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-05-01 21:37  NiLuJe

	* [r17174] src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libharfbuzz.so.0, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-04-06 04:25  NiLuJe

	* [r17028] src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-03-10 15:28  NiLuJe

	* [r16952] src/linkss/bin/convert, src/linkss/bin/identify,
	  src/linkss/bin/inotifywait, src/linkss/bin/mogrify,
	  src/linkss/bin/shlock, src/linkss/bin/sort,
	  src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libharfbuzz.so.0,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libpng16.so.16,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-02-11 03:08  NiLuJe

	* [r16858] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Refresh binaries (fix gdb)

2020-02-10 16:58  NiLuJe

	* [r16851] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2020-02-04 18:01  NiLuJe

	* [r16835] src/linkss/bin/convert, src/linkss/bin/identify,
	  src/linkss/bin/inotifywait, src/linkss/bin/mogrify,
	  src/linkss/bin/shlock, src/linkss/bin/sort,
	  src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libharfbuzz.so.0,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libpng16.so.16,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-11-25 22:32  NiLuJe

	* [r16737] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Update binaries

2019-11-23 23:12  NiLuJe

	* [r16715] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libharfbuzz.so.0, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-11-14 15:43  NiLuJe

	* [r16658] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libharfbuzz.so.0,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libturbojpeg.so.0,
	  src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-10-11 21:21  NiLuJe

	* [r16572] src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-10-06 19:37  NiLuJe

	* [r16535] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-10-01 02:33  NiLuJe

	* [r16517] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libharfbuzz.so.0,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libturbojpeg.so.0:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-09-15 22:01  NiLuJe

	* [r16474] src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libjpeg.so.62,
	  src/linkss/lib/libturbojpeg.so.0, src/linkss/lib/libz.so.1:
	  
	  Kindle Hacks:
	  * Refresh binaries

2019-09-02 11:12  NiLuJe

	* [r16438] src/linkss/bin/convert, src/linkss/bin/identify,
	  src/linkss/bin/inotifywait, src/linkss/bin/mogrify,
	  src/linkss/bin/shlock, src/linkss/bin/sort,
	  src/linkss/etc/ImageMagick-6/delegates.xml,
	  src/linkss/lib/libMagickCore-6.Q8.so.6,
	  src/linkss/lib/libMagickWand-6.Q8.so.6,
	  src/linkss/lib/libfreetype.so.6, src/linkss/lib/libharfbuzz.so.0,
	  src/linkss/lib/libjpeg.so.62, src/linkss/lib/libpng16.so.16,

src/uninstall.sh
#!/bin/sh
#
# ScreenSavers uninstaller
#
# $Id: uninstall.sh 11231 2014-12-07 14:22:33Z NiLuJe $
#
##

HACKNAME="linkss"

# Pull libOTAUtils for logging & progress handling
[ -f ./libotautils5 ] && source ./libotautils5


otautils_update_progressbar

# Remove our deprecated content
# From simple_screensaver
logmsg "I" "uninstall" "" "removing deprecated files (simple_screensaver)"
# v1
if [ -f /etc/upstart/screensaver_hack.conf ] ; then
    rm -f /etc/upstart/screensaver_hack.conf
fi
# v2
if [ -f /etc/upstart/custom_screensaver.conf ] ; then
	rm -f /etc/upstart/custom_screensaver.conf
fi
if [ -f /usr/lib/blanket/custom_screensaver.so ] ; then
	# Bundle this here, since I have no idea how resilient blanket is,
	# and how well it would deal with trying to unload a non-existent module.
	lipc-set-prop com.lab126.blanket unload custom_screensaver
	rm -f /usr/lib/blanket/custom_screensaver.so
fi

otautils_update_progressbar

# Delete upstart job
logmsg "I" "uninstall" "" "removing upstart job"
if [ -f /etc/upstart/${HACKNAME}.conf ] ; then
    rm -f /etc/upstart/${HACKNAME}.conf
fi

otautils_update_progressbar

# Remove custom directory in userstore?
logmsg "I" "uninstall" "" "removing kual extension (only if /mnt/us/${HACKNAME}/uninstall exists)"
if [ -d /mnt/us/extensions/${HACKNAME} -a -f /mnt/us/${HACKNAME}/uninstall ] ; then
    rm -rf /mnt/us/extensions/${HACKNAME}
    logmsg "I" "uninstall" "" "kual extension has been removed"
fi
logmsg "I" "uninstall" "" "removing custom directory (only if /mnt/us/${HACKNAME}/uninstall exists)"
if [ -d /mnt/us/${HACKNAME} -a -f /mnt/us/${HACKNAME}/uninstall ] ; then
    rm -rf /mnt/us/${HACKNAME}
    logmsg "I" "uninstall" "" "custom directory has been removed"
fi

otautils_update_progressbar

logmsg "I" "uninstall" "" "done"

otautils_update_progressbar

return 0


src/install.sh
#!/bin/sh
#
# ScreenSavers installer
#
# $Id: install.sh 16252 2019-07-23 21:44:39Z NiLuJe $
#
##

HACKNAME="linkss"

# Pull libOTAUtils for logging & progress handling
[ -f ./libotautils5 ] && source ./libotautils5


otautils_update_progressbar

# Install our hack's custom content
# Check which default screensaver to use...
kmodel="$(cut -c3-4 /proc/usid)"
case "${kmodel}" in
    "13" | "54" | "2A" | "4F" | "52" | "53" )
        # Voyage...
        logmsg "I" "install" "" "using the default voyage custom screensaver"
        HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me.png ${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
    ;;
    "24" | "1B" | "1D" | "1F" | "1C" | "20" | "D4" | "5A" | "D5" | "D6" | "D7" | "D8" | "F2" | "17" | "60" | "F4" | "F9" | "62" | "61" | "5F" )
        # PaperWhite...
        logmsg "I" "install" "" "using the default paperwhite custom screensaver"
        HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me.png ${HACKNAME}/screensavers/00_you_can_delete_me-kv.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
    ;;
    "0F" | "11" | "10" | "12" | "C6" | "DD" )
        # Touch & KT2
        logmsg "I" "install" "" "using the default touch custom screensaver"
        HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-kv.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
    ;;
    * )
        # Try the new device ID scheme...
        kmodel="$(cut -c4-6 /proc/usid)"
        case "${kmodel}" in
            "0G1" | "0G2" | "0G4" | "0G5" | "0G6" | "0G7" | "0KB" | "0KC" | "0KD" | "0KE" | "0KF" | "0KG" | "0LK" | "0LL" | "0GC" | "0GD" | "0GR" | "0GS" | "0GT" | "0GU" | "0PP" | "0T1" | "0T2" | "0T3" | "0T4" | "0T5" | "0T6" | "0T7" | "0TJ" | "0TK" | "0TL" | "0TM" | "0TN" | "102" | "103" | "16Q" | "16R" | "16S" | "16T" | "16U" | "16V" )
                # PW3/PW4 & Oasis...
                logmsg "I" "install" "" "using the default paperwhite 3/4 & oasis custom screensaver"
                HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me.png ${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
            ;;
            "0DU" | "0K9" | "0KA" | "10L" | "0WF" | "0WG" | "0WH" | "0WJ" | "0VB" )
                # KT3/KT4...
                logmsg "I" "install" "" "using the default touch custom screensaver"
                HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-kv.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
            ;;
            "0LM" | "0LN" | "0LP" | "0LQ" | "0P1" | "0P2" | "0P6" | "0P7" | "0P8" | "0S1" | "0S2" | "0S3" | "0S4" | "0S7" | "0SA" | "11L" | "0WQ" | "0WP" | "0WN" | "0WM" | "0WL" )
                # Oasis 2/Oasis 3...
                logmsg "I" "install" "" "using the default oasis 2/3 custom screensaver"
                HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me.png ${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-kv.png"
            ;;
            * )
                # Fallback
                logmsg "I" "install" "" "using the default touch custom screensaver"
                HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-kv.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
            ;;
        esac
    ;;
esac

otautils_update_progressbar

# But keep the user's custom content...
if [ -d /mnt/us/${HACKNAME} ] ; then
    logmsg "I" "install" "" "our custom directory already exists, checking if we have custom content to preserve"
    # Custom Screensavers
    if [ "x$( ls -A /mnt/us/${HACKNAME}/screensavers 2> /dev/null )" != x ] ; then
        # If we already have a non-empty custom ss dir, exclude the default custom ss from our archive
        HACK_EXCLUDE="${HACKNAME}/screensavers/00_you_can_delete_me.png ${HACKNAME}/screensavers/00_you_can_delete_me-pw.png ${HACKNAME}/screensavers/00_you_can_delete_me-kv.png ${HACKNAME}/screensavers/00_you_can_delete_me-koa2.png"
        logmsg "I" "install" "" "found custom screensavers, excluding default custom screensaver"
    fi
    # If we disabled autoreboot, don't re-enable it...
    if [ ! -f /mnt/us/${HACKNAME}/autoreboot ] ; then
        HACK_EXCLUDE="${HACK_EXCLUDE} ${HACKNAME}/autoreboot"
        logmsg "I" "install" "" "keep autoreboot status (off)"
    fi
fi

otautils_update_progressbar

# Okay, now we can extract it. Since busybox's tar is very limited, we have to use a tmp directory to perform our filtering
logmsg "I" "install" "" "installing custom directory"
# Make sure our xzdec binary is executable first...
chmod +x ./xzdec
./xzdec ${HACKNAME}.tar.xz | tar -xvf -
# Do check if that went well
_RET=$?
if [ ${_RET} -ne 0 ] ; then
    logmsg "C" "install" "code=${_RET}" "failed to extract custom directory in tmp location"
    return 1
fi

otautils_update_progressbar

cd src
# And now we filter the content to preserve user's custom content
for custom_file in ${HACK_EXCLUDE} ; do
    if [ -f "./${custom_file}" ] ; then
        logmsg "I" "install" "" "preserving custom content (${custom_file})"
        rm -f "./${custom_file}"
    fi
done
# Finally, unleash our filtered dir on the live userstore
cp -af . /mnt/us/
_RET=$?
if [ ${_RET} -ne 0 ] ; then
    logmsg "C" "install" "code=${_RET}" "failure to update userstore with custom directory"
    return 1
fi
cd - >/dev/null
rm -rf src

otautils_update_progressbar

# Remove our deprecated content
# From simple_screensaver
logmsg "I" "install" "" "removing deprecated files (simple_screensaver)"
# v1
if [ -f /etc/upstart/screensaver_hack.conf ] ; then
    rm -f /etc/upstart/screensaver_hack.conf
fi
# v2
if [ -f /etc/upstart/custom_screensaver.conf ] ; then
	rm -f /etc/upstart/custom_screensaver.conf
fi
if [ -f /usr/lib/blanket/custom_screensaver.so ] ; then
	# Bundle this here, since I have no idea how resilient blanket is,
	# and how well it would deal with trying to unload a non-existent module.
	lipc-set-prop com.lab126.blanket unload custom_screensaver
	rm -f /usr/lib/blanket/custom_screensaver.so
fi

otautils_update_progressbar

# From v0.6.N
logmsg "I" "install" "" "removing deprecated files (mobi_unpack)"
if [ -f /mnt/us/${HACKNAME}/bin/mobi_unpack.py ] ; then
    rm -f /mnt/us/${HACKNAME}/bin/mobi_unpack.py
fi
if [ -f /mnt/us/${HACKNAME}/bin/mobi_unpack.pyc ] ; then
    rm -f /mnt/us/${HACKNAME}/bin/mobi_unpack.pyc
fi
if [ -f /mnt/us/${HACKNAME}/bin/mobi_unpack.pyo ] ; then
    rm -f /mnt/us/${HACKNAME}/bin/mobi_unpack.pyo
fi

otautils_update_progressbar

# From v0.8.N
logmsg "I" "install" "" "removing deprecated files (old imagemagick config)"
if [ -d /mnt/us/${HACKNAME}/etc/ImageMagick ] ; then
    rm -rf /mnt/us/${HACKNAME}/etc/ImageMagick
fi

otautils_update_progressbar


# Setup startup script
logmsg "I" "install" "" "installing upstart job"
cp -f ${HACKNAME}.conf /etc/upstart/${HACKNAME}.conf

otautils_update_progressbar

# Generating Python bytecode for Mobi Unpack
logmsg "I" "install" "" "generating python bytecode"
# We need python, of course ;)
if [ -f "/mnt/us/python/bin/python2.7" ] ; then
    /mnt/us/python/bin/python2.7 -m py_compile /mnt/us/${HACKNAME}/bin/kindleunpack.py /mnt/us/${HACKNAME}/bin/mobi_uncompress.py /mnt/us/${HACKNAME}/bin/path.py /mnt/us/${HACKNAME}/bin/utf8_utils.py /mnt/us/${HACKNAME}/bin/kfxmeta.py
    /mnt/us/python/bin/python2.7 -O -m py_compile /mnt/us/${HACKNAME}/bin/kindleunpack.py /mnt/us/${HACKNAME}/bin/mobi_uncompress.py /mnt/us/${HACKNAME}/bin/path.py /mnt/us/${HACKNAME}/bin/utf8_utils.py /mnt/us/${HACKNAME}/bin/kfxmeta.py
fi

otautils_update_progressbar

logmsg "I" "install" "" "cleaning up"
rm -f ${HACKNAME}.conf ${HACKNAME}.tar.xz xzdec

logmsg "I" "install" "" "done"

otautils_update_progressbar

return 0
